examples/dnsblock/dnsblock.lua

```diff
@@ -0,0 +1,89 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- Filter DNS packets based on a blocklist
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local blacklist = {
+	"github.com",
+	"gitlab.com",
+}
+
+local function nop() end
+
+local function check_blacklist (name)
+	for _,v in ipairs(blacklist) do
+		if string.find(name, v) ~= nil then
+			return true
+		end
+	end
+	return false
+end
+
+local function get_domainname (skb, off)
+	local name = ""
+	local i = 0
+	while true do
+		local d = skb:getbyte(off + i)
+		if d == 0 then
+			break
+		end
```

Comment:
I'm not very fond of loops without proper ending conditions.. you could change this to `repeat ... until d == 0` right?

Reply:
btw, perhaps it's a better idea to just find the last position (e.g., `d == 0`) and then convert the string as a whole, using `skb:getstring(off, length)`

Reply:
The issue here is that, DNS uses question name of the following format :
`<length1><prefix1><length2><prefix2><length3><prefix3>....<lengthN><prefixN><00>`

Example : 
`\x03www\x06google\x03com\x00`

I think `repeat..until d == 0` will be a better choice

Reply:
`repeat..until d == 0`, will also have an issue, 

```lua
	repeat
		d = skb:getbyte(off + i)
		if d == 0 then
			break
		end
		...
```

I still need this condition `d == 0` check to prevent the loop reading further bytes

Reply:
`repeat..until d == 0`, will also have an issue, 

```lua
	repeat
		d = skb:getbyte(off + i)
		if d == 0 then
			break
		end
		...
```

I still need this condition `d == 0` check to prevent the loop reading further bytes

Reply:
> The issue here is that, DNS uses question name of the following format : `<length1><prefix1><length2><prefix2><length3><prefix3>....<lengthN><prefixN><00>`
> 
> Example : `\x03www\x06google\x03com\x00`

you can just use gsub later on, right?

Reply:
btw, instead of looking for `\0`, checking byte per byte, you could just transform the rest of the payload into a string and apply pattern matching using string facilities (e.g., `match`, `gsub`), right? wouldn't that be a simpler approach?